### PR TITLE
[posix-host] parse all parameters to a structure

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit parses all parameters to a structure and puts all parameters parsing code in the function `ParseArg()`. It makes us easier to add new parameters to posix host.

### CODE REVIEW

File: src/posix/platform/openthread-system.h [hunk 1]
```diff
@@ -90,12 +90,17 @@ enum
...
-    const char *mRadioConfig;   /// Radio configurations.
-    uint32_t    mSpeedUpFactor; /// Speed up factor.
-    bool        mResetRadio;    /// Whether to reset RCP when initializing.
+    uint32_t    mSpeedUpFactor;     ///< Speed up factor.
+    uint8_t     mLogLevel;          ///< Debug level of logging.
+    bool        mIsDryRun;          ///< Dry run.
+    bool        mPrintRadioVersion; ///< Whether to print radio firmware version.
+    bool        mIsVerbose;         ///< Whether to print log to stderr.
+
+    uint64_t    mNodeId;        ///< Unique node ID.
```
Comment (bukepo) [inline src/posix/platform/openthread-system.h]: what's consideration to move this down?

File: src/posix/platform/openthread-system.h [hunk 2]
```diff
@@ -90,12 +90,17 @@ enum
...
 typedef struct otPlatformConfig
 {
-    uint64_t    mNodeId;        /// Unique node ID.
-    const char *mInterfaceName; /// Thread network interface name.
-    const char *mRadioFile;     /// Radio file path.
-    const char *mRadioConfig;   /// Radio configurations.
-    uint32_t    mSpeedUpFactor; /// Speed up factor.
-    bool        mResetRadio;    /// Whether to reset RCP when initializing.
+    uint32_t    mSpeedUpFactor;     ///< Speed up factor.
+    uint8_t     mLogLevel;          ///< Debug level of logging.
```
Comment (bukepo) [inline src/posix/platform/openthread-system.h]: I think `otLogLevel` is better.

File: src/posix/platform/openthread-system.h [hunk 3]
```diff
@@ -90,12 +90,17 @@ enum
...
+    bool        mIsDryRun;          ///< Dry run.
+    bool        mPrintRadioVersion; ///< Whether to print radio firmware version.
```
Comment (bukepo) [inline src/posix/platform/openthread-system.h]: Why put `IsDryRun` and `PrintRadioVersion` here? They are really temporary necessary.

File: src/posix/platform/openthread-system.h (hunk 2 above)
Comment (zhanglongxia) [reply src/posix/platform/openthread-system.h]: Updated

File: src/posix/platform/openthread-system.h (hunk 3 above)
Comment (zhanglongxia) [reply src/posix/platform/openthread-system.h]: Just want to put all parsed parameters in s single structure. I have updated the code to use another method to process this problem.

File: src/posix/platform/openthread-system.h (hunk 1 above)
Comment (zhanglongxia) [reply src/posix/platform/openthread-system.h]: I just want to put system configurations together and put radio configurations together. It seems not necessary . Will update it.

File: src/posix/main.c [hunk 4]
```diff
@@ -94,19 +103,13 @@ static void PrintUsage(const char *aProgramName, FILE *aStream, int aExitCode)
     exit(aExitCode);
 }
 
-static otInstance *InitInstance(int aArgCount, char *aArgVector[])
+void ParseArg(int aArgCount, char *aArgVector[], PosixConfig *aConfig)
```
Comment (bukepo) [inline src/posix/main.c]: keep `static`?

Comment (zhanglongxia) [reply src/posix/main.c]: Updated

File: src/posix/platform/hdlc_interface.hpp [hunk 5]
```diff
@@ -103,16 +103,14 @@ class HdlcInterface
      *
      * @note This method should be called before reading and sending frames to the interface.
      *
-     *
-     * @param[in]   aRadioFile    The path to either a UART device or an executable.
-     * @param[in]   aRadioConfig  Parameters to be given to the device or executable.
+     * @param[in]  aPlatformConfig  Argument vector.
```
Comment (jwhui) [inline src/posix/platform/hdlc_interface.hpp]: nit: what about "Platform configuration structure"?

Comment (zhanglongxia) [reply src/posix/platform/hdlc_interface.hpp]: Updated.

**Pull request merged**